### PR TITLE
updates for node 6

### LINKED ISSUE #370: bcrypt is breaking login.wmo on windows
bcrypt requires node-gyp, node-gyp requires an amazing VC++ stack, bottom line: brcypt can't install and login.wmo won't run without it.

Is there a quick path to success here?

### LINKED ISSUE #382: sqlite and jshint are breaking the build
sqlite needs a C++ '11 compiler, and jshint doesn't work with the older version of the grunt plugin.

#379 fixes this.

### PR DESCRIPTION
**Pull request by Pomax**
fixes to sqlite3 and routes.js so that login can actually run on modern node (6+).

Oddly the fix in routes.js is for code that should not even work to begin with, because the env var that gets read in is a String, whereas the code pretend it's an array. Not sure why that didn't cause errors before, but it sure caused errors now =D

fixes #370, fixes #382

### CODE REVIEW

**Comment by Pomax** [comment]
@cadecairos R?

File: app/http/routes.js [hunk 1]
```diff
@@ -12,11 +12,19 @@ module.exports = function (http, modelsController, webmakerAuth) {
     },
     userList = env.get("ALLOWED_USERS"),
     authMiddleware = basicAuth(function (user, pass) {
-      for (var username in userList) {
-        if (userList.hasOwnProperty(username)) {
-          if (user === username && pass === userList[username]) {
-            return true;
-          }
+      if (typeof userList === "string") {
```
Comment (cadecairos) [inline app/http/routes.js]: how could this have ever worked? is habitat doing something magical?

Comment (Pomax) [inline app/http/routes.js]: I have **no** idea

Comment (cadecairos) [inline app/http/routes.js]: nope, id definitely communicates with login using basic auth

Comment (Pomax) [inline app/http/routes.js]: in that case I am drawing a blank :confused:

**Comment by Pomax** [comment]
compile errors during build fixed using the gcc solution found in https://sudogem.wordpress.com/2016/06/14/travis-ci-this-version-of-nodenanv8-requires-a-c11-compiler/

**Comment by Pomax** [comment]
Ugh, `Warning: Path must be a string. Received null Use --force to continue` are you bloody kidding me?

**Comment by Pomax** [comment]
OMG MY GO WHY IS THIS SO HARD. @cadecairos looks like things work...

File: app/http/routes.js [hunk 2]
```diff
@@ -12,11 +12,20 @@ module.exports = function (http, modelsController, webmakerAuth) {
...
+        var arrList = {};
+        var extractUserPass = function (pair) {
+          var terms = pair.split(":");
+          arrList[terms[0]] = terms[1];
+        };
+        userList.split(",").forEach(extractUserPass);
+        userList = arrList;
+      }
+      var keys = Object.keys(userList);
```
Comment (Pomax) [inline app/http/routes.js]: we could make this an Object.keys(userList).some(...) but no real reason to I think..

**Comment by cadecairos** [comment]
LGTM

**Pull request merged**